### ISSUE TITLE
Form elements not available when form detached from DOM [regression in 3.6.0]

### ISSUE THREAD

**Issue by zbynek** [HtmlUnit/htmlunit#740]
The following prints a single input element to console in Chrome:
```
const form = document.createElement("form");
form.appendChild(document.createElement("input"));
form.remove();
console.log(form.elements);
```
but in HtmlUnit `form.elements` becomes empty once form is detached from DOM.

Seems to come from https://github.com/HtmlUnit/htmlunit/commit/00ce1ab2d0c546ba2b20c38730643ca94912b8dc#diff-ccb37821109ece9c771414e5408dc628f9baf4e80e0ad5c5d16f93e5576ef2a6R609

If the form is detached it would make sense to iterate over its children instead of the whole page.

**Comment by rbri** [comment]
Great catch, will have a look. Thanks

**Comment by rbri** [comment]
again thanks for the report, this is now fixed in the repo.
Will make a new snapshot build available soon - this will be a 4.0.0-SNAPSHOT because of the major impact the removal of the IE support has to the codebase.
The current plan for the release is end march/start april - hopt that is fine for you

**Comment by zbynek** [comment]
Thanks for the fast fix. The workaround for this is quite simple so there's no rush as far as I am concerned.
